### ISSUE TITLE
Link Files for Image

### ISSUE THREAD

**Issue by oliverstorm** [contao/core#5863]
Would be good to be able to choose file for Image.
As Sample i have an image and want to link it to an pdf in files. At the moment i can only choose a page but not a file to be linked by image. 
At the moment this link must be set manually.
![filepicker-image](https://f.cloud.github.com/assets/4511789/590275/4919a66c-c9dc-11e2-84e0-463ce7d321ec.png)

**Comment by mrflory** [comment]
This issue still bothers me (and my customers). In tinyMCE the switch from file to site chooser is already implemented. Can this be done for the link and download elements as well?

**Comment by mrflory** [comment]
I came up with the following workaround: I created a `dcaconfig.php` file in the config folder with the following code:
```php
if (!class_exists('tl_dcaconfig')) {
	$GLOBALS['TL_DCA']['tl_content']['fields']['url']['eval']['filesOnly'] = true;
	$GLOBALS['TL_DCA']['tl_content']['fields']['url']['wizard'][] = array('tl_dcaconfig', 'filePicker');

	class tl_dcaconfig
	{
		/**
		 * Return the file picker wizard
		 * @param \DataContainer
		 * @return string
		 */
		public function filePicker(DataContainer $dc)
		{
			return ' <a href="contao/file.php?do='.Input::get('do').'&amp;table='.$dc->table.'&amp;field='.$dc->field.'&amp;value='.$dc->value.'" title="'.specialchars(str_replace("'", "\\'", $GLOBALS['TL_LANG']['MSC']['filepicker'])).'" onclick="Backend.getScrollOffset();Backend.openModalSelector({\'width\':768,\'title\':\''.specialchars($GLOBALS['TL_LANG']['MOD']['files'][0]).'\',\'url\':this.href,\'id\':\''.$dc->field.'\',\'tag\':\'ctrl_'.$dc->field . ((Input::get('act') == 'editAll') ? '_' . $dc->id : '').'\',\'self\':this});return false">' . Image::getHtml('pickfile.gif', $GLOBALS['TL_LANG']['MSC']['filepicker'], 'style="vertical-align:top;cursor:pointer"') . '</a>';
		}
	}
}
```
This adds a file picker button next to the page picker button. Any improvements are welcome.

**Comment by leofeyer** [comment]
Fixed in contao/core-bundle@46ee448c2c53fd186f789fb6fa64150ad0360777. Needs to be back ported.

**Comment by leofeyer** [comment]
Back ported in 18010b506ceab46742747145faabe20c0a89b77a.

**Comment by leofeyer** [comment]
Fixed the other pickers, too, in b2641f1946314ecaf4039f394381d6f12eda6e7b.
